src/Providers/Maps/Google/Features/Shape.ts

```diff
@@ -0,0 +1,147 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+namespace Provider.Maps.Google.Feature {
+    export class Shape
+        implements
+            OSFramework.Maps.Feature.IShape,
+            OSFramework.Maps.Interface.IBuilder,
+            OSFramework.Maps.Interface.IDisposable
+    {
+        private _isInsideShape: boolean;
+        private _map: OSFramework.Maps.OSMap.IMap;
+        private _markerCoordinates;
+        private _returnObjCode: string;
+        private _returnObjMessage: string;
+        private _returnObjSuccess: boolean;
+        private _shape: OSFramework.Maps.Shape.IShape;
+
+        // Method for shape circle exception
+        private _shapeCircle(): void {
+            const circleRadius = this._shape.provider.getRadius();
+            const circleCenter = this._shape.provider.getCenter();
+            const distanceBetweenPoints =
+                google.maps.geometry.spherical.computeDistanceBetween(
+                    this._markerCoordinates,
+                    circleCenter
+                );
+
+            this._isInsideShape = distanceBetweenPoints <= circleRadius;
+            this._returnObjSuccess = true;
+            this._returnObjCode = OSFramework.Maps.Enum.Success.code;
+            this._returnObjMessage = this._isInsideShape.toString();
+        }
+
+        // Method to apply the default calculations for shapes
+        private _shapeDefault(): void {
+            this._isInsideShape = google.maps.geometry.poly.containsLocation(
+                this._markerCoordinates,
+                this._map.getShape(this._shape.widgetId).provider
+            );
+            this._returnObjSuccess = true;
+            this._returnObjCode = OSFramework.Maps.Enum.Success.code;
+            this._returnObjMessage = this._isInsideShape.toString();
+        }
+
+        // Method for shape rectangle exception
+        private _shapeRectangle(): void {
+            this._isInsideShape = this._map
+                .getShape(this._shape.widgetId)
+                .provider.getBounds()
+                .contains(this._markerCoordinates);
+            this._returnObjSuccess = true;
+            this._returnObjCode = OSFramework.Maps.Enum.Success.code;
+            this._returnObjMessage = this._isInsideShape.toString();
+        }
+
+        public build(): void {
```

Comment:
We can delete this after delete the implementation.

Reply:
Done

Reply:
Done